### DISCUSSION TITLE
[Script request] Maintainerr

### DISCUSSION THREAD

**Discussion by Washday101** [community-scripts/ProxmoxVE#97] [Request script]
### Application Name

Maintainerr

### Website

https://maintainerr.info/

### Description

Maintainerr makes managing your media easy. Maintainerr is a way to systematically remove media from your server that is taking up space. You can set rules and Maintainerr will create a collection of media items that match your rules, and set them up for future deletion.

### Due Diligence

- [X] I have searched existing [scripts](https://community-scripts.github.io/Proxmox/scripts) and found no duplicates.
- [X] I have searched existing [discussions](https://github.com/community-scripts/ProxmoxVE/discussions?discussions_q=) and found no duplicate requests.
- [X] This is not a game-related request.

**Comment by Bishop-trevorstuart** [comment]
I too would like to see this as a script.

**Comment by nfribeiro** [comment]
That would be great

**Comment by WildEchoWanderer** [comment]
it would be awesome

i made this installer script but don't have access to my proxmox test-setup to test it. if anybody wants to test this: (i have no clue if it works or if i have done it correctly)

bash -c "$(wget -qLO - https://raw.githubusercontent.com/Andurilll/ProxmoxVE/refs/heads/main/install/maintainerr-install.sh)"

**Comment by caroipdev** [comment]
~~I've opened a PR to add the script https://github.com/community-scripts/ProxmoxVE/pull/652~~

I'm still testing locally

**Reply by WildEchoWanderer** [reply]
wow what a fast progress

i am new to coding/github. may you tell me how do you know what files to add (ct\maintainerr.sh, install\maintainerr-install.sh & json\maintainerr.json) ?
i knew i need to specify resources ("cpu": 2, "ram": 2048, ...) but i don't see this in other scripts unter install\___-install.sh (like overseerr) written out. where is the link to the resources?

**Reply by caroipdev** [reply]
I'm also new to this project, if you have a look to the latest accepted PRs https://github.com/community-scripts/ProxmoxVE/pulls?q=is%3Apr+label%3A%22new+script%22+is%3Amerged 

You can see all of them add 3 files, then you call follow the lead. 

Your suggestion:
> bash -c "$(wget -qLO - https://raw.githubusercontent.com/Andurilll/ProxmoxVE/refs/heads/main/install/maintainerr-install.sh)"

Does not work because maintainerr requires a build process, I've based my approach with a few ct/*.sh, eg: the-lounge-install.sh, hoarder-install.sh and magicmirror-install.sh.

**Reply by benscobie** [reply]
Hi, I'm a contributor for Maintainerr. We can definitely make this script a lot simpler. Currently we only build for Docker but I see no reason why we can't build artifacts for platforms like linux-x64. I'll see if I can make some time to look at this.

With the script the way it is, there's a high chance it'll break in the future as we move things around.

**Reply by caroipdev** [reply]
@benscobie that would be great, it would simplify quite a lot and also reduce the memory needed, currently it need 2GB to make sure yarn does not crash because of lack of memory.

It would be nice to have the built files replacing the process.env.* instead of needing to be provided at the runtime.

**Comment by MickLesk** [comment]
Update:
The developers at Maintainerr are currently working on a linux*64 solution. The script is currently on hold.

**Reply by WildEchoWanderer** [reply]
Just for me to understand: what is Different with a Linux*x64 Version?

**Reply by MickLesk** [reply]
its a “ready-made” file that actually contains everything you need. Without any migration steps or adjustments here and there.

**Comment by Wicksy24** [comment]
Any updates on a this?

**Reply by MickLesk** [reply]
Nope

**Reply by benscobie** [reply]
The Maintainner side of things is in-progress, albeit slow as I've been working on other projects: https://github.com/jorenn92/Maintainerr/pull/1562

**Reply by MickLesk** [reply]
@benscobie I have the PR on my watch list, I can see it's progressing quite well. Do you now have an approximate estimate of when it will be ready?

**Comment by tumeden** [comment]
Hey,   was about to post a request for this and found this thread.   Would love to have a script to auto install Maintainerr.

Reading the discussion here,  any updates on how this is coming along?

**Comment by MickLesk** [comment]
@jorenn92 I'll just ping you directly :smile:

You mentioned an x64 Linux variant some time ago, there was also a PR. What happened to it?

**Reply by MickLesk** [reply]
Ah sorry i meant @benscobie :smile:

**Reply by benscobie** [reply]
[The PR](https://github.com/jorenn92/Maintainerr/pull/1562) is still open and is sat with @jorenn92 to review

**Reply by tumeden** [reply]
Would love to see it implemented.  A lot of proxmox users that would love to install maintainerr into their *arr ecosystem.    Cheers.

**Comment by rc5kamps** [comment]
Just found Maintainerr as well (even though it really should be spelled "Maintainarr" (staying with the *Arr suite of programs) :) 
Either way, I just wanted to say I would love to see a Maintainerr/MaintainArr LCX for Proxmox.   As I really don't want to have to install Docker VM on my Proxmox :)

**Reply by MickLesk** [reply]
We wait for this merge:
https://github.com/Maintainerr/Maintainerr/pull/1562

**Comment by artogahr** [comment]
Related / alternative I requested, which is focused on Jellyfin instead of Plex: https://github.com/community-scripts/ProxmoxVE/discussions/10350

**Comment by trinitron79** [comment]
+1 on this in a LXC

**Comment by appleimperio** [comment]
Now that Maintainerr has support for Jellyfin. Please make it happen!

**Comment by Kugelstift** [comment]
@MickLesk seems to be closed ? What needs to happen now ?

**Reply by MickLesk** [reply]
Idk. Its Not our Project. @benscobie can you give Feedback please?

**Reply by fran1987** [reply]
> Idk. Its Not our Project. @benscobie can you give Feedback please?


This [commit](https://github.com/Maintainerr/Maintainerr/commit/4b2d96c14be5fdf90890ead30e6a205170cc0b22) does not look promising  :(

**Reply by MickLesk** [reply]
Strange Project

**Comment by appleimperio** [comment]
Warning. I just received an email closing this issue as completed with a link to a clone Proxmox scripts page https://heretek-ai.github.io/ProxmoxVE/scripts?id=maintainerr

**Reply by MickLesk** [reply]
This is Not us. Copycat

**Comment by AdmiralAwesome** [comment]
Script got recently updated: https://github.com/Maintainerr/Maintainerr/releases/tag/v3.8.0
Maybe it's time to get it here?

**Comment by MickLesk** [comment]
https://github.com/community-scripts/ProxmoxVED/issues/2041
